vcd/resource_vcd_service_account_test.go

```diff
@@ -0,0 +1,284 @@
+	configText2 := templateFill(testAccServiceAccount_SysOrgDS, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 2: %s", configText2)
+
+	params["FuncName"] = t.Name() + "step3"
+	configText3 := templateFill(testAccServiceAccount_SysOrg_Active, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 2: %s", configText3)
+
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	t.Cleanup(deleteApiTokenFile(t.Name()))
+	resourceName := "vcd_service_account.sysadmin"
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: testAccProviders,
+		CheckDestroy:      testAccCheckServiceAccountDestroy(params["Org"].(string), params["SaName"].(string)),
+		Steps: []resource.TestStep{
+			{
+				Config: configText1,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", t.Name()),
+					resource.TestCheckResourceAttr(resourceName, "software_id", params["SoftwareId"].(string)),
+					resource.TestCheckResourceAttr(resourceName, "software_version", params["SoftwareVersion"].(string)),
+					resource.TestCheckResourceAttr(resourceName, "uri", params["Uri"].(string)),
+					resource.TestCheckResourceAttr(resourceName, "active", "false"),
+				),
+			},
+			{
+				Config: configText2,
+				Check:  resourceFieldsEqual("data.vcd_service_account.sysadmin_ds", resourceName, []string{"file_name", "allow_token_file", "%"}),
+			},
+			{
+				Config: configText3,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", t.Name()),
+					resource.TestCheckResourceAttr(resourceName, "software_id", params["SoftwareIdUpdated"].(string)),
+					resource.TestCheckResourceAttr(resourceName, "software_version", params["SoftwareVersionUpdated"].(string)),
+					resource.TestCheckResourceAttr(resourceName, "uri", params["UriUpdated"].(string)),
+					resource.TestCheckResourceAttr(resourceName, "active", "true"),
+					testCheckFileExists(params["FileName"].(string)),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateIdFunc:       importStateIdOrgObject(params["Org"].(string), params["SaName"].(string)),
+				ImportStateVerifyIgnore: []string{"org", "file_name", "allow_token_file"},
+			},
+		},
+	})
+}
+
+const testAccServiceAccount_SysOrg_Role = `
+data "vcd_role" "sys_admin" {
+  org  = "{{.Org}}"
+  name = "{{.RoleName}}"
+}
+`
+
+const testAccServiceAccount_SysOrg = testAccServiceAccount_SysOrg_Role + `
+resource "vcd_service_account" "sysadmin" {
+  name = "{{.SaName}}"
+  org  = "{{.Org}}"
+
+  role             = data.vcd_role.sys_admin.id
+  software_id      = "{{.SoftwareId}}"
+  software_version = "{{.SoftwareVersion}}"
+  uri              = "{{.Uri}}"
+
+  active = false
+}
+`
+
+const testAccServiceAccount_SysOrgDS = testAccServiceAccount_SysOrg + `
+# skip-binary-test: datasource test will fail when run together with resource
```

Comment:
This data source can run in binary tests if we add a dependency on the corresponding resource
```hcl
data "vcd_service_account" "sysadmin_ds" {
  name = vcd_service_account.sysadmin.name
  org  = vcd_service_account.sysadmin.org
}
```

Reply:
Updated.